packages/node/src/runtimePlugin.ts

```diff
@@ -150,48 +151,74 @@ export default function () {
               }
             });
         }
+        function httpVmStrategy(
+          chunkName: string,
+          remoteName: string,
+          callback: (err: Error | null, chunk: any) => void,
+        ): void {
+          const http = __non_webpack_require__('http');
+          const https = __non_webpack_require__('https');
+          const vm = __non_webpack_require__('vm');
 
-        function httpVmStrategy(chunkName, remoteName, callback) {
-          var http = __non_webpack_require__('http');
-          var https = __non_webpack_require__('https');
-          var vm = __non_webpack_require__('vm');
-
-          var url = resolveUrl(remoteName, chunkName);
+          const url = resolveUrl(remoteName, chunkName);
           if (!url) {
-            var emptyChunk = {
+            const emptyChunk = {
               modules: {}, // No modules
               ids: [], // No chunk IDs
               runtime: null, // No runtime function
             };
             return callback(null, emptyChunk);
           }
-          var protocol = url.protocol === 'https:' ? https : http;
-          protocol.get(url.href, function (res) {
-            var data = '';
-            res.on('data', function (chunk) {
+
+          const fetchMethod =
+            globalThis.webpackChunkLoad ||
+            (url.protocol === 'https:' ? https : http).get;
```

Comment:
```suggestion
            __non_webpack_require__(url.protocol === 'https:' ? 'https' : 'http').get;
```

Reply:
I would avoid importing both http and https if just one of them is actually used

Reply:
perfect